### ISSUE TITLE
I have thousands of labeled images, any value to train on that many ?

### ISSUE THREAD

**Issue by Xqua** [MouseLand/cellpose#140]
I am working on volumetric light sheet datasets, and I have one fully annotated embryo across 300 time points. 
Each time point has from 8 cell to 1200 cells at the end (so no slice ever has more than a few dozen cells).

CellPose "out of the box" nuclei model works "okay" but it is missing quite a few nuclei, so I am thinking of retraining the network using my dataset. 
LightSheet microscopy does produce images that are quite different looking compared to "regular" microscopy. Especially there is a "large" amount of artifact signal due to postprocessing ("comet tail")

I can generate as many slices as I want from this dataset. 

What would be a "decent" amount of segmented images to feed the network? 1000? 10000? 100000? 
Is there value in number vs quality? Should I cherry-pick images where the network fails to detect nuclei instead of brute-forcing it?

**Comment by kevinjohncutler** [comment]
I've gotten excellent results with about 1000 cells spread over 3 images. Furthermore, it only took a few much smaller images to train on slightly different images (e.g. structured noise from the camera sensor, optical artifacts, etc. from a colleague's different imaging setup). You can start small and scale up as your hardware allows (google colab may be a good option if that is a limiting factor for you). I personally got better results training the network from scratch, but you should try training from the pretrained model as well. It is reasonable to assume that there is a trade-off of accuracy for generalization.

I would cherry-pick only as far as to make sure the training set is as representative as possible of the data you expect to segment in the future. I usually keep a couple of the most problematic images aside so that I can evaluate the model on a worst-case scenario (but that assumes there are equally bad images in the training dataset, of course).

**Comment by Xqua** [comment]
Thanks @kevinjohncutler  for your anwser ! 

So it would seem that mass is not needed for this network :D

Then I'll extract slices as they come comparing the set to false positive and negatives to try to get towards generalizing my set !
